src/test/java/seedu/address/logic/commands/ListPrioCommandTest.java

```diff
@@ -0,0 +1,111 @@
+                new PriorityMatchesPredicate(highPriority);
+
+        ListPrioCommand listPrioFirstCommand = new ListPrioCommand(firstPredicate);
+        ListPrioCommand listPrioSecondCommand = new ListPrioCommand(secondPredicate);
+
+        // same object -> returns true
+        assertTrue(listPrioFirstCommand.equals(listPrioFirstCommand));
+
+        // same values -> returns true
+        ListPrioCommand listPrioCommandCopy = new ListPrioCommand(firstPredicate);
+        assertTrue(listPrioFirstCommand.equals(listPrioCommandCopy));
+
+        // different types -> returns false
+        assertFalse(listPrioFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(listPrioFirstCommand.equals(null));
+
+        // different person -> returns false
+        assertFalse(listPrioFirstCommand.equals(listPrioSecondCommand));
+    }
+
+    @Test
+    public void execute_highPriority_success() {
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 2);
+        PriorityMatchesPredicate predicate = new PriorityMatchesPredicate(highPriority);
+        ListPrioCommand command = new ListPrioCommand(predicate);
+        expectedModel.updateFilteredPersonList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(ALICE, BENSON), model.getFilteredPersonList());
+    }
+
+    @Test
+    public void execute_lowPriority_success() {
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 2);
+        PriorityMatchesPredicate predicate = new PriorityMatchesPredicate(lowPriority);
+        ListPrioCommand command = new ListPrioCommand(predicate);
+        expectedModel.updateFilteredPersonList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(CARL, DANIEL), model.getFilteredPersonList());
+    }
+
+    @Test
+    public void execute_mediumPriority_success() {
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 2);
+        PriorityMatchesPredicate predicate = new PriorityMatchesPredicate(mediumPriority);
+        ListPrioCommand command = new ListPrioCommand(predicate);
+        expectedModel.updateFilteredPersonList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(ELLE, FIONA), model.getFilteredPersonList());
+    }
+
+    @Test
+    public void execute_nonePriority_success() {
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 1);
+        PriorityMatchesPredicate predicate = new PriorityMatchesPredicate(noPriority);
+        ListPrioCommand command = new ListPrioCommand(predicate);
+        expectedModel.updateFilteredPersonList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(GEORGE), model.getFilteredPersonList());
+    }
```

Comment:
Test cases are comprehensive and checks every possible scenario for `Priority` 👍🏻